app/controllers/VerifiedEmailController.scala

```diff
@@ -53,6 +55,23 @@ class VerifiedEmailController @Inject() (
     }
   }
 
+  def getVerifiedEmailV2(): Action[AnyContent] = authorisedRequest async {
+    implicit request: RequestWithEori[AnyContent] =>
+      emailRepo.get(request.eori.value).flatMap {
+        case Some(value) => Future.successful(Ok(Json.toJson(value)))
+        case None        => retrieveAndStoreEmail(request.eori.value)
+      }
+  }
+
+  private def retrieveAndStoreEmail(eori: String): Future[Result] =
```

Comment:
Better to keep this private method after all the public methods as public methods are the ones that should be on the top